test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperLabelExtensionsTest.cs

```diff
@@ -252,6 +252,38 @@ public void LabelFor_DisplaysVariableName()
             Assert.Equal("<label for=\"HtmlEncode[[unknownKey]]\">HtmlEncode[[unknownKey]]</label>", HtmlContentUtilities.HtmlContentToString(result));
         }
 
+        [Fact]
+        public void LabelHelpers_UsesSpecifiedLabelText()
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper();
+
+            // Act
+            var labelResult = helper.Label("Property1", labelText: "Hello");
+            var labelForResult = helper.LabelFor(m => m.Property1, labelText: "Hello");
+            var labelForModelResult = helper.LabelForModel(labelText: "Hello");
+
+            // Assert
+            Assert.Equal("<label for=\"HtmlEncode[[Property1]]\">HtmlEncode[[Hello]]</label>", HtmlContentUtilities.HtmlContentToString(labelResult));
+            Assert.Equal("<label for=\"HtmlEncode[[Property1]]\">HtmlEncode[[Hello]]</label>", HtmlContentUtilities.HtmlContentToString(labelForResult));
+            Assert.Equal("<label for=\"\">HtmlEncode[[Hello]]</label>", HtmlContentUtilities.HtmlContentToString(labelForModelResult));
+        }
+
+        [Fact]
+        public void LabelHelpers_DisplaysSpecifiedHtmlAttributes()
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper();
+
+            // Act
+            var labelForResult = helper.LabelFor(m => m.Property1, htmlAttributes: new { attr="value" });
+            var labelForModelResult = helper.LabelForModel(labelText: "Hello", htmlAttributes: new { attr="value" });
```

Comment:
Split these two different scenarios. Again, `LabelFor(m => m)` is the call that's analogous to `LabelForModel()` but both are special cases unless you're in a partial, `EditorFor()`, or similar.